pkg/chunk/delete_store.go

```diff
@@ -0,0 +1,366 @@
+package chunk
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"flag"
+	"fmt"
+	"hash/fnv"
+	"strings"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+const (
```

Comment:
I am not sure what advantage that would have except have smaller names for each status type.
I prefer this because it the names have enough context to be used outside of the package

Reply:
Generally named types make it easier to understand a function signature. I know you are using these as part of the index client. However, I don't think you should be doing that. I think the delete client should have a separate interface with a function that does something along the lines of `SetStatus(ctx context.Context, planName string, status PlanStatus) error`

Reply:
Enums in general make understanding the code much easier. You can immediately understand what options you have when using a function by finding the named type. When you use a `string` it makes it much harder to understand what options are provided by the API. It also makes it much easier to misuse.